src/networkmanager.cpp

```diff
@@ -161,11 +161,28 @@ void NetworkManager::handle_server_packet()
     }
   }
 
-  const QStringList packet_list = in_data.split("%", QString::SkipEmptyParts);
+  const QStringList packet_list = in_data.split("%", Qt::SkipEmptyParts);
```

Comment:
Not sure how well this handles the case where the following packets come in:

`123#%45678`
`901#%23456`
...

Because there is always a partial message in each packet, no message is processed and the buffer can be filled until it results in a crash.

Here is an alternate implementation (looks like you did it in less lines): https://github.com/AttorneyOnline/AO2-Client/blob/1949012bdb661167910dad362ce153c9a6341bb7/src/network/legacysocket.cpp#L29-L62

Reply:
"Partial Packet" above handles this